pkgs/os-specific/linux/kernel/cachyos-kernels.nix

```diff
@@ -0,0 +1,312 @@
+      NO_HZ_COMMON = yes;
+      CONTEXT_TRACKING = yes;
+
+      # Preempt type
+      PREEMPT_BUILD = yes;
+      PREEMPT_NONE = unset;
+      PREEMPT_VOLUNTARY = lib.mkOverride 60 unset;
+      PREEMPT = lib.mkOverride 60 yes;
+      PREEMPT_COUNT = yes;
+      PREEMPTION = yes;
+      PREEMPT_DYNAMIC = yes;
+
+      # Enable O3
+      CC_OPTIMIZE_FOR_PERFORMANCE = unset;
+      CC_OPTIMIZE_FOR_PERFORMANCE_O3 = yes;
+
+      # Enable bbr3
+      TCP_CONG_CUBIC = lib.mkOverride 60 module;
+      DEFAULT_CUBIC = unset;
+      TCP_CONG_BBR = yes;
+      DEFAULT_BBR = yes;
+      DEFAULT_TCP_CONG = freeform "bbr";
+
+      # Select THP
+      TRANSPARENT_HUGEPAGE_MADVISE = lib.mkOverride 60 unset;
+      TRANSPARENT_HUGEPAGE_ALWAYS = lib.mkOverride 60 yes;
+
+      USER_NS = yes;
+
+      # Arch Linux RCU config
+      TREE_RCU = yes;
+      PREEMPT_RCU = yes;
+      RCU_EXPERT = yes;
+      TREE_SRCU = yes;
+      TASKS_RCU_GENERIC = yes;
+      FORCE_TASKS_RCU = unset;
+      NEED_TASKS_RCU = yes;
+      TASKS_RCU = yes;
+      FORCE_TASKS_RUDE_RCU = unset;
+      TASKS_RUDE_RCU = yes;
+      FORCE_TASKS_TRACE_RCU = unset;
+      TASKS_TRACE_RCU = yes;
+      RCU_STALL_COMMON = yes;
+      RCU_NEED_SEGCBLIST = yes;
+      RCU_FANOUT = freeform "64";
+      RCU_FANOUT_LEAF = freeform "16";
+      RCU_BOOST = yes;
+      RCU_BOOST_DELAY = freeform "500";
+      RCU_EXP_KTHREAD = unset;
+      RCU_NOCB_CPU = yes;
+      RCU_NOCB_CPU_DEFAULT_ALL = unset;
+      RCU_NOCB_CPU_CB_BOOST = unset;
+      TASKS_TRACE_RCU_READ_MB = unset;
+      RCU_LAZY = yes;
+      RCU_LAZY_DEFAULT_OFF = yes;
+      RCU_DOUBLE_CHECK_CB_TIME = yes;
+
+      # Modules
+      ## Apple T2 https://github.com/t2linux/linux-t2-patches/blob/main/extra_config
+      APPLE_BCE = module;
+      APPLE_GMUX = module;
+      BRCMFMAC = module;
+      BT_BCM = module;
+      BT_HCIBCM4377 = module;
+      BT_HCIUART_BCM = yes;
+      BT_HCIUART = module;
+      HID_APPLETB_BL = module;
+      HID_APPLETB_KBD = module;
+      HID_APPLE = module;
+      DRM_APPLETBDRM = module;
+      HID_SENSOR_ALS = module;
+      SENSORS_APPLESMC = module;
+      SND_PCM = module;
+      STAGING = yes;
+
+      I2C_NCT6775 = module;
+      NTSYNC = module;
+      V4L2_LOOPBACK = module;
+    };
+  };
+
+  cachyosKernelFor =
+    {
+      version,
+      variantDescription,
+      source,
+      patches,
+      config,
+    }:
+    buildLinux (
+      args
+      // rec {
+        inherit version;
+        pname = "linux-cachyos";
+        modDirVersion = version;
+        isCachy = true;
+
+        src = pkgs.fetchurl {
+          url = source.url;
+          hash = source.hash;
+        };
+
+        kernelPatches = patches;
+
+        structuredExtraConfig = config;
+
+        extraMeta = {
+          branch = lib.versions.majorMinor version;
+          maintainers = with lib.maintainers; [ drakon64 ];
+          description = variantDescription;
+          broken = stdenv.isAarch64;
```

Comment:
Instead of setting it as `meta.broken`, remove it from `meta.platforms`.

Reply:
Though, I think this could work on `aarch64-linux`, it's just that this has not been tested.

Reply:
It might work, but I'm currently not in a position to test it

Reply:
@PedroHLC Do you have an example of doing this? Both Zen and XanMod use `broken = stdenv.isAarch64;`.

Reply:
setting `broken` is fine for now.

Reply:
> setting `broken` is fine for now.

No, there is a BIG difference here: Zen supports Aarch64, it is "broken in Nixpkgs" -- while CachyOS themselves does not support Aarch64.

In theory, by removing it from "platforms" should also remove it from documentation/search for that platform. But I guess that's not being used by search.nixos.org or `legacyPackages`.